src/mlir/error.h

```diff
@@ -19,49 +19,78 @@
 // They all must have a description. Parsing related errors must also have a
 // file location, pointing to the original place the failing function is
 // trying to work with. Stacking multiple errors generate a stack trace.
+//
+// FIXME: The descriptions are initialised by the caller directly with s
+// string value, which doesn't work when the error could be emitted in
+// multiple languages. We need a mechanism where strings could come from
+// any language with a formatted output for a list of arguments, filled in
+// by the caller.
 namespace mlir::verona
 {
   // RuntimeError: Errors that have no relation to the source file
   class RuntimeError : public llvm::ErrorInfo<RuntimeError>
   {
+    // Description of the error
     std::string desc;
 
   public:
+    // Internal ID to control matching
     static char ID;
     RuntimeError(llvm::StringRef desc) : desc(desc) {}
+
+    // Public ErrorInfo interface, requires raw_ostream to output
+    // the string in the right way. This is used by error handlers.
     void log(llvm::raw_ostream& OS) const override
     {
       OS << desc;
     }
+    // Public ErrorInfo interface, converts to error_code for
+    // interoperation with other error handling mechanisms.
+    // For now, we don't care about that, but once we start working
+    // with other error handling we should implement this for real.
     std::error_code convertToErrorCode() const override
     {
       return llvm::inconvertibleErrorCode();
     }
   };
-  // Create a parsing error and return
+
+  /// Create and return a new runtime error.
+  /// The returned error will own a copy of the string passed as `desc`.
```

Comment:
This is the only case where the doc comments are actually doc comments.  I'd prefer if all of them were.  LLVM typically uses `///`, but support for that in editors is a bit unreliable.  I generally perfer:
```c
/**
 * Some comment
 */
```
It uses more vertical space, but also gives a clear visual indicator.